privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardPayloadAdapter.kt

```diff
@@ -34,6 +35,11 @@ class AppPrivacyDashboardPayloadAdapter @Inject constructor(
         val payloadAdapter = moshi.adapter(Payload::class.java)
         return kotlin.runCatching { payloadAdapter.fromJson(payload)?.url ?: "" }.getOrDefault("")
     }
+    override fun onOpenSettings(payload: String): String {
```

Comment:
what's inside `payload`? it seems nothing we use. I see we are mapping the payload into `SettingsPayload`, but when opening the screen, no value is required. Maybe we don't need any logic here?

Reply:
Looking closely, I think @shakyShane's original intention was to make this method generic, and the `payload` would clarify which particular setting we'd like to open. So I [changed the logic](https://github.com/duckduckgo/Android/pull/2738/commits/7203f8b8fca575228c4bc5323f99a50a97c0fe83) a bit. This also addresses the your remarks about naming